File: gamedex-datamodel/src/main/java/com/github/ykrasik/gamedex/datamodel/persistence/Library.java
package com.github.ykrasik.gamedex.datamodel.persistence;

import com.github.ykrasik.gamedex.datamodel.GamePlatform;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.Value;

import java.nio.file.Path;

/**
 * @author Yevgeny Krasik
 */
@Value
@Builder
@EqualsAndHashCode(of = "id")
public class Library implements PathEntity, Comparable<Library> {
    @NonNull private final Id<Library> id;
    @NonNull private final String name;
    @NonNull private final Path path;
    @NonNull private final GamePlatform platform;

    @Override
    public int compareTo(Library o) {
        return name.compareTo(o.name);
    }
}

File: gamedex-datamodel/src/main/kotlin/com/github/ykrasik/gamedex/datamodel/persistence/Library.kt
package com.github.ykrasik.gamedex.datamodel.persistence

import com.github.ykrasik.gamedex.datamodel.GamePlatform
import java.nio.file.Path

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 11:29
 */
data class Library(
    val id: Int,
    val path: Path,
    val platform: GamePlatform,
    val name: String
) : Comparable<Library> {

    override fun compareTo(other: Library) = name.compareTo(other.name)

    override fun toString() = name
}